internal/statesync/syncer.go

```diff
@@ -38,12 +38,16 @@ var (
 	errRejectFormat = errors.New("snapshot format was rejected")
 	// errRejectSender is returned by Sync() when the snapshot sender is rejected.
 	errRejectSender = errors.New("snapshot sender was rejected")
-	// errVerifyFailed is returned by Sync() when app hash or last height verification fails.
+	// errVerifyFailed is returned by Sync() when app hash or last height
+	// verification fails.
 	errVerifyFailed = errors.New("verification failed")
 	// errTimeout is returned by Sync() when we've waited too long to receive a chunk.
 	errTimeout = errors.New("timed out waiting for chunk")
 	// errNoSnapshots is returned by SyncAny() if no snapshots are found and discovery is disabled.
 	errNoSnapshots = errors.New("no suitable snapshots found")
+	// errDeadlineExceeded is returned by Sync() when the timeout for retrieving
+	// tendermint state or the commit is exceeded
+	errDeadlineExceeded = errors.New("deadline to retrieve state and commit exceeded")
```

Comment:
This already is an error value in the standard library, can we just use that one?

Reply:
The reason why I made a separate error is what happens if someone uses a deadline in the parent context. If it gets triggered we have (AFAIK and context is relatively new to me so I don't know too much) no idea whether the deadline was from the parent context or the sub context we use for retrieving the state and commit information. Does that make sense?

Reply:
I see that this used on line 287 and 295 of this file. In either of those cases, the parent or the child could have timed out and we sill return this error. We still can't tell which one timed out when we precolate this error up but perhaps I'm not totally understanding?

Reply:
Ok good catch. I think I've added the adequate clause to distinguish between parent context errors and child context errors